File: dehaze-go/internal/model/sys_announcement.go
package model

import "time"

// SysAnnouncement 系统公告表
type SysAnnouncement struct {
	BaseModel
	Title        string     `gorm:"column:title;type:varchar(255);default:'';comment:公告标题" json:"title"`
	Content      string     `gorm:"column:content;type:text;not null;comment:公告内容" json:"content"`
	Type         string     `gorm:"column:type;type:varchar(32);not null;default:operation;comment:公告类型" json:"type"`
	Importance   int8       `gorm:"column:importance;type:tinyint;not null;default:1;comment:重要级别(1:普通;2:重要)" json:"importance"`
	TargetScope  string     `gorm:"column:target_scope;type:varchar(32);not null;default:all;comment:发送范围" json:"targetScope"`
	TargetParams string     `gorm:"column:target_params;type:json;comment:范围参数JSON" json:"targetParams"`
	Status       int8       `gorm:"column:status;type:tinyint;not null;default:1;comment:公告状态(1:草稿;2:待发送;3:已发送;4:已取消)" json:"status"`
	SendTime     *time.Time `gorm:"column:send_time;type:datetime;comment:发送时间" json:"sendTime"`
	ExpireTime   *time.Time `gorm:"column:expire_time;type:datetime;comment:过期时间" json:"expireTime"`
	SentCount    int        `gorm:"column:sent_count;type:int;default:0;comment:已发送人数" json:"sentCount"`
	Deleted      int8       `gorm:"column:deleted;type:tinyint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysAnnouncement) TableName() string {
	return "sys_announcement"
}

File: dehaze-python/app/models/entity/sys_announcement.py
from datetime import datetime
from typing import Any

from sqlalchemy import JSON, BigInteger, DateTime, String, Text
from sqlalchemy.dialects import mysql as mysql_types
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysAnnouncement(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_announcement"
    __table_args__ = {"comment": "系统公告表"}

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    title: Mapped[str] = mapped_column(String(255), nullable=False, default="", comment="公告标题")
    content: Mapped[str] = mapped_column(Text, nullable=False, comment="公告内容")
    type: Mapped[str] = mapped_column(
        String(32), nullable=False, default="operation", comment="公告类型"
    )
    importance: Mapped[int] = mapped_column(
        mysql_types.TINYINT, nullable=False, default=1, comment="重要级别(1:普通;2:重要)"
    )
    target_scope: Mapped[str] = mapped_column(
        String(32), nullable=False, default="all", comment="发送范围"
    )
    target_params: Mapped[Any | None] = mapped_column(JSON, nullable=True, comment="范围参数")
    status: Mapped[int] = mapped_column(
        mysql_types.TINYINT,
        nullable=False,
        default=1,
        comment="公告状态(1:草稿;2:待发送;3:已发送;4:已取消)",
    )
    send_time: Mapped[datetime | None] = mapped_column(DateTime, nullable=True, comment="发送时间")
    expire_time: Mapped[datetime | None] = mapped_column(
        DateTime, nullable=True, comment="过期时间"
    )
    sent_count: Mapped[int | None] = mapped_column(nullable=False, default=0, comment="已发送人数")
